src/gpio.rs

```diff
@@ -86,20 +93,18 @@ fn irq_handler(port_wakers: &[Option<&PortWaker>]) {
     let reg = unsafe { crate::pac::Gpio::steal() };
 
     for (port, port_waker) in port_wakers.iter().enumerate() {
-        if port_waker.is_none() {
-            continue;
-        }
-
-        let stat = reg.intstata(port).read().bits();
-        for pin in BitIter(stat) {
-            // Clear the interrupt from this pin
-            reg.intstata(port).write(|w| unsafe { w.status().bits(1 << pin) });
-            // Disable interrupt from this pin
-            reg.intena(port)
-                .modify(|r, w| unsafe { w.int_en().bits(r.int_en().bits() & !(1 << pin)) });
-
-            if let Some(waker) = port_waker.unwrap().get_waker(pin as usize) {
-                waker.wake();
+        if let Some(port_waker) = port_waker {
```

Comment:
Then this extra conditional can be removed, because you guarantee that you will only iterate through `Some(AtomicWaker)`.